lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php

```diff
@@ -449,7 +449,7 @@ protected static function append_to_selector( $selector, $to_append, $position =
 	 *
 	 * @return array Block metadata.
 	 */
-	protected static function get_blocks_metadata() {
```

Comment:
This stands out as something problematic to me. Consumers should not use the private classes. See https://github.com/WordPress/gutenberg/pull/45168 for a comprehensive list of issues I see with the current implementation of the theme.json APIs. The most relevant here are the sections about creating public API when necessary and do not leak private implementation details.

What's the data structure you need from a plugin point of view? A map from the block name (`core/paragraph`) to its CSS selector (`p`)? There may be an opportunity to make this public differently (creating a public method for this operation).

Reply:
That is exactly what we need! We would be happy to help create that public method for this operation.

Reply:
I've noted this conversation in a couple of threads, for us to keep an eye on. https://github.com/WordPress/gutenberg/issues/45171 and https://github.com/WordPress/gutenberg/issues/45194#issuecomment-1300685965

Would you be able to create a pull request that does this only thing (create this public API method)? Ideally, it'd take a single block name and will return its selector. It could live in `lib/compat/wordpress-6.2/get-global-styles-and-settings.php`.

Reply:
Thanks for breaking that down so well @oandregal! I've put up [this PR](https://github.com/WordPress/gutenberg/pull/45505) to expose this public method. 

I'll update this PR to remove the change to `blocks_metadata`

Reply:
I've removed the change I made to that method!